lib/rules/no-typos.js

```diff
@@ -54,38 +57,69 @@ module.exports = {
       }
     }
 
+    function isPropTypesPackage(node) {
+      // Note: we really do want == with the package names here,
+      // since we need value equality, not identity - and
```

Comment:
i'm confused about this - every string that has the same contents is the same string by identity, iow `'abc' === 'abc'`. Why do we need `==` here?

Reply:
I thought that too, but when I was testing, it seemed that only every string *literal* with the same contents is the same string by identity. I found that was not the case when comparing a literal to a value read from a file.

But maybe there was another issue that I didn't fully understand; I made [this jsfiddle](https://jsfiddle.net/3yu2op32/5/) to test this, and the results indicate no difference between `===` and `==`.

I can dig into this more tomorrow, probably.

Reply:
Perhaps there's invisible characters in the strings? When you run into the difference, try comparing `.length` (altho then `==` would fail too). All strings in JS that have the same contents are `===`.

Specifically, `==` is "`===` after type coercion", so if they're both `typeof "string"`, `==` and `===` are identical, by spec.

Reply:
WOW ok silly mistake. `propTypesPackageName` was an array containing one string, which is indeed `==` to that string value itself, but not `===`.

Thanks for not letting that slide. 😳 Pushing fix shortly.

![](https://memegenerator.net/img/instances/65251662/javascript.jpg)